kong/core/handler.lua

```diff
@@ -117,14 +119,12 @@ return {
           "' with: "..tostring(err))
       end
 
-      if balancer_address.hostname and not api.preserve_host then
-        var.upstream_host = balancer_address.hostname
+      -- if set `host_header` is the original header to be preserved
+      var.upstream_host = host_header or 
```

Comment:
_nitpick_ not using else is a more defensive programming practice

```
var.upstream_host = balancer_address.hostname..":"..balancer_address.port
if host_header then
    var.upstream_host = host_header
end
```

reference: https://github.com/TheLadders/object-calisthenics#rule-2-dont-use-the-else-keyword